### DISCUSSION TITLE
Volatility Stop

### DISCUSSION THREAD

**Discussion by rudeGit** [DaveSkender/Stock.Indicators#564] [Indicators]
This library does not seem to have volatility stop indicator as described on the BarChart web site (https://www.barchart.com/education/technical-indicators/volatility-stop). If I am wrong, please let me know.

The explanation for this indicator seemed simple enough so that I can compute this indicator myself. However the very first step _(i.e. Determine the initial trend direction)_ stopped me cold. Should I use EMA based computation for the initial trend? Or should I use SMA? Or should I use ADX? What should I use? Or am I thinking this in the wrong way?

Any help is greatly appreciated.

**Comment by DaveSkender** [comment]
It’s not uncommon for indicators to have a warm-up window that contains throwaway data. This indicator is very similar to Welles Wilder’s [Parabolic SAR](https://daveskender.github.io/Stock.Indicators/indicators/ParabolicSar/#content) that we initialize with the assumption that the initial trend is upward/bullish.  You could guess wrong, but the good news is even if you guess wrong it will auto-correct itself after the first pivot; which is why we always exclude that first warmup period from results.  It's also very similar to [SuperTrend](https://daveskender.github.io/Stock.Indicators/indicators/SuperTrend/#content).

**Reply by DaveSkender** [reply]
And you're right, we don’t have this in the library, so I’ve added it to the backlog.  If your interested in contributing this new indicator to the library, mention your intention on #565

**Reply by rudeGit** [reply]
Thanks for the reply. I am still figuring out how to work with the library when ever I get some time. I might be able to contribute once I am comfortable with it.

**Reply by DaveSkender** [reply]
This was implemented in version 1.18.0
and uses the original formula presented in Welles Wilder’s book [New Concepts in Technical Trading Systems](https://archive.org/details/newconceptsintec00wild).
